### ISSUE TITLE
Fallback Datasource Item not retrieved in Experience Editor

### ISSUE THREAD

**Issue by jo3stevens** [mikeedwards83/Glass.Mapper#375]
I'm working on a multilingual site where everything work on the fallback language.

When I create a new language version of a page containing multiple components with datasource items attached the page works perfectly fine on the front end with the datasource showing the fallback language versions.

If I open the new language version in Experience Editor every component fails to get it's datasource item. The page contains View and Controller renderings both of which are failing.

Debugging one of my Controller renderings I have code like this:

if (_mvcContext.HasDataSource)
{
	return View(_sitecoreService.GetItem<Tile_Carousel>(_mvcContext.DataSourceItem));
}

mvcContent.DataSourceItem has retrieved the datasource item successfully, but sitecoreService.GetItem returns null rather than getting the fallback item. This is only happening when in Experience Editor. Debugging the same code from the front end retrieves the item.

If there something I am missing to have this working in EE?

**Comment by mikeedwards83** [comment]
Hi

I have just setup this as a test and I see the fallback language in the EE editor, in both edit and preview mode. 

I am using V5.

I can look at reviewing your solution, for more info see our support page:
http://www.glass.lu/Support

Mike

**Comment by jo3stevens** [comment]
Hi Mike

Just to give some further detail which may help.

I'm using Glass Mapper 5.2.13 and Sitecore 9.0.2

My fallback language is en, and when the page itself and a rendering on that page using a datasource item that also has a en language version everything is okay.

I then create a new language version of the page, but not the datasource item, so the page may now be en-au, but the datasource item still only has the en language. This is where EE is falling over. If I go to my datasource item and create an en-au language version of it, then EE will work as expected.

I have item and field level fallback enabled globally. I may try upgrading to the latest version of Glass, but if you still can't reproduce I may need to arrange some support.

**Comment by jo3stevens** [comment]
Hi @mikeedwards83 

We found the reason for this behavior; it was due to the slightly unconventional way we attempted to use language fallback.

The idea was that the base language, 'en' would never actually be used to render the site, and would just be used for fallback content. We had changed the language in the Sites config to use another language by default, and that was what was causing the problem.

I'm assuming that Glass was perhaps attempting to get the language version for the language set in the Sites config which didn't have a version and that was causing the problem we saw.

Regardless it's a non-issue now as we've reverted to a conventional use of fallback.

Cheers
Joe
